examples/marketdata/marketdata.go

```diff
@@ -120,6 +122,53 @@ func cryptoQuote() {
 	fmt.Println()
 }
 
+func optionChain() {
+	chain, err := marketdata.GetOptionChain("AAPL", marketdata.GetOptionSnapshotRequest{})
+	if err != nil {
+		panic(err)
+	}
+	type snap struct {
+		marketdata.OptionSnapshot
+		Symbol string
+	}
+	calls, puts := []snap{}, []snap{}
+	for symbol, snapshot := range chain {
+		if strings.Contains(symbol, "C") {
```

Comment:
This can be misleading if the user requests a symbol, that contains a letter 'C'. This is possibly too complex, but shows the idea:
```go
idx := len(symbol) - 9
if idx < 7 { // shortest symbol: A123456C12345678
    panic("invalid symbol")
}
switch (symbol[idx]) {
    case 'C':
        calls = append(calls, snap{OptionSnapshot: snapshot, Symbol: symbol})
    case 'P':
        puts = append(puts, snap{OptionSnapshot: snapshot, Symbol: symbol})
    default:
        panic("invalid symbol") 
}
```

Reply:
This is just an example. The user cannot "request a symbol", AAPL is hardcoded.